File: app/src/main/java/com/gameautopilot/app/brain/BrainFactory.kt
package com.gameautopilot.app.brain

import com.gameautopilot.app.data.BrainProvider
import com.gameautopilot.app.data.Settings

object BrainFactory {
    fun create(settings: Settings): Brain {
        val baseUrl = settings.baseUrl.ifBlank { Settings.defaultUrlFor(settings.provider) }
        val model = settings.model.ifBlank { Settings.defaultModelFor(settings.provider) }
        return when (settings.provider) {
            BrainProvider.GEMINI -> GeminiBrain(baseUrl = baseUrl, apiKey = settings.apiKey, model = model)
            BrainProvider.OPENAI, BrainProvider.NVIDIA ->
                OpenAiCompatibleBrain(baseUrl = baseUrl, apiKey = settings.apiKey, model = model)
        }
    }
}

File: mac/Sources/GameAutopilotMac/Brain/BrainFactory.swift
import Foundation

enum BrainFactory {
    static func create(settings: Settings, apiKey: String) -> Brain {
        switch settings.provider {
        case .gemini:
            return GeminiBrain(baseUrl: settings.resolvedBaseUrl, apiKey: apiKey, model: settings.resolvedModel)
        case .openai, .nvidia:
            return OpenAICompatibleBrain(baseUrl: settings.resolvedBaseUrl, apiKey: apiKey, model: settings.resolvedModel)
        }
    }
}
